src/chem/aeronet_aaod2ioda.py

```diff
@@ -133,83 +125,106 @@ def add_data(infile):
+    varAttrs[('surface_type', 'MetaData')]['units'] = ''
 
     # Whether aaod reaches Level 2.0 without the threshold of aod440 >= 0.4 (0: yes, 1: no)
-    loc_mdata['aaod_l2_qc_without_aod440_le_0.4_threshold'] = np.where(f3['if_retrieval_is_l2(without_l2_0.4_aod_440_threshold)'] == 1, 0, 1)
+    outdata[('aaod_l2_qc_without_aod440_le_0.4_threshold', 'MetaData')] = np.where(f3['if_retrieval_is_l2(without_l2_0.4_aod_440_threshold)'] == 1, 0, 1)
+    varAttrs[('aaod_l2_qc_without_aod440_le_0.4_threshold', 'MetaData')]['units'] = ''
 
     # Whether Coincident_AOD440nm in aeronet_cad.txt reaches Level 2.0 (0: yes, 1: no)
-    loc_mdata['aod_l2_qc'] = np.where(f3['if_aod_is_l2'] == 1, 0, 1)
+    outdata[('aod_l2_qc', 'MetaData')] = np.where(f3['if_aod_is_l2'] == 1, 0, 1)
+    varAttrs[('aod_l2_qc', 'MetaData')]['units'] = ''
 
     # aaod inversion type: 0 for ALM20 and 1 for ALM15
-    loc_mdata['aaod_l2_qc'] = np.where(f3['inversion_data_quality_level'] == 'lev20', 0, 1)
+    outdata[('aaod_l2_qc', 'MetaData')] = np.where(f3['inversion_data_quality_level'] == 'lev20', 0, 1)
+    varAttrs[('aaod_l2_qc', 'MetaData')]['units'] = ''
 
-    c = np.empty([nlocs], dtype='S50')
+    c = np.empty([nlocs], dtype=object)
     c[:] = np.array(f3.siteid)
-    loc_mdata['station_id'] = writer.FillNcVector(c, 'string')
+    outdata[('station_id', 'MetaData')] = c
+    varAttrs[('station_id', 'MetaData')]['units'] = ''
 
-    # Define datetime
-    d = np.empty([nlocs], 'S20')
+    d = np.empty([nlocs], dtype=object)
     for i in range(nlocs):
         d[i] = f3.time[i].strftime('%Y-%m-%dT%H:%M:%SZ')
-    loc_mdata['datetime'] = writer.FillNcVector(d, 'datetime')
-
-    # Define var_mdata
-    var_mdata['frequency'] = writer.FillNcVector(frequency, 'float')
-    var_mdata['sensor_channel'] = writer.FillNcVector(aeronetinv_chan, 'integer')
-
-    for key, value in obsvars.items():
-        outdata[varDict[key]['valKey']] = np.array(f3[value].fillna(nc.default_fillvals['f4']))
-        outdata[varDict[key]['qcKey']] = np.where(outdata[varDict[key]['valKey']] == nc.default_fillvals['f4'],
-                                                  1, 0)
-        if key in ["aerosol_optical_depth_3", "aerosol_optical_depth_5", "aerosol_optical_depth_6", "aerosol_optical_depth_7"]:
-            outdata[varDict[key]['errKey']] = np.where(outdata[varDict[key]['valKey']] == nc.default_fillvals['f4'],
-                                                       nc.default_fillvals['f4'], 0.02)
-        else:
-            outdata[varDict[key]['errKey']] = np.array(nc.default_fillvals['f4'], dtype=np.float32)
-
-    # Define global atrributes
-    AttrData = {'observation_type': 'AERONET AAOD',
-                'sensor': "aeronet",
-                'surface_type': 'ocean=0, land=1, costal=2'}
+    outdata[('datetime', 'MetaData')] = d
+    varAttrs[('datetime', 'MetaData')]['units'] = ''
+
+    outdata[('frequency', 'VarMetaData')] = np.float32(frequency)
```

Comment:
Please place all metadata variables in the group "MetaData". We have deprecated the "VarMetaData" group. Thanks!